riptide-spring-boot-starter/src/main/java/org/zalando/riptide/spring/DefaultRiptideRegistrar.java

```diff
@@ -394,15 +395,19 @@ private Tag clientName(final String id, final Client client) {
     }
 
     private BeanMetadataElement trace(final String executor) {
-        if (registry.isRegistered("tracer")) {
-            return genericBeanDefinition(TracingExecutors.class)
-                    .setFactoryMethod("preserve")
-                    .addConstructorArgReference(executor)
-                    .addConstructorArgReference("tracer")
-                    .getBeanDefinition();
-        } else {
-            return ref(executor);
-        }
+        final AtomicReference<BeanMetadataElement> result = new AtomicReference<>();
+
+        ifPresent("org.zalando.tracer.concurrent.TracingExecutors", () -> {
```

Comment:
Maybe it's prettier to expose a method `Dependencies#isPresent(String)` that simply returns a Boolean

Reply:
I used the supplier version since that was easier to cover. But having both cases is already impossible to cover since you can't change the classpath during tests...